client/checker_client/checker.py

```diff
@@ -294,14 +336,28 @@ def deploy_wtez(
     ttl: Optional[int] = None,
 ):
     print("Deploying the wtez contract.")
+    config = load_input_config()
     src = repo.wtez_contract
+
+    with repo.wtez_metadata.open() as f:
+        views = json.load(f)["views"]
+
+    # Compute the TZIP-16 metadata
+    token_metadata_view = wtez_token_metadata_view_from_config(config=config)
+    metadata = tzip16_metadata_from_views(views + [token_metadata_view])
+    metadata_ser = json.dumps(metadata).encode("utf-8")
```

Comment:
Out of curiosity, is storing the metadata as a JSON string part of the FA2 spec?

Reply:
Indeed it is! See the ["Contract Storage" section of TZIP16](https://gitlab.com/tezos/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#contract-storage):

> At least one value must be present:
> * the one for the empty string key (`""`).
> * the value must be a URI as specified in the following section which points to a JSON document as specified further below.
> [..]